pkg/db/db_deployments.go

```diff
@@ -0,0 +1,655 @@
+	"fmt"
+	"github.com/freiheit-com/kuberpult/pkg/logger"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+	"time"
+)
+
+type DBDeployment struct {
+	Created        time.Time
+	ReleaseVersion *int64
+	App            string
+	Env            string
+	TransformerID  TransformerID
+	Metadata       string // json
+}
+
+type DeploymentMetadata struct {
+	DeployedByName  string
+	DeployedByEmail string
+	CiLink          string
+}
+
+type Deployment struct {
+	Created       time.Time
+	App           string
+	Env           string
+	Version       *int64
+	Metadata      DeploymentMetadata
+	TransformerID TransformerID
+}
+
+// SELECT
+
+func (h *DBHandler) DBSelectLatestDeployment(ctx context.Context, tx *sql.Tx, appSelector string, envSelector string) (*Deployment, error) {
+	span, ctx := tracer.StartSpanFromContext(ctx, "DBSelectLatestDeployment")
+	defer span.Finish()
+	selectQuery := h.AdaptQuery(`
+		SELECT created, releaseVersion, appName, envName, metadata, transformereslVersion
+		FROM deployments
+		WHERE appName=? AND envName=?
+		LIMIT 1;
+	`)
+	span.SetTag("query", selectQuery)
+	rows, err := tx.QueryContext(
+		ctx,
+		selectQuery,
+		appSelector,
+		envSelector,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("could not select deployment from DB. Error: %w\n", err)
+	}
+	defer func(rows *sql.Rows) {
+		err := rows.Close()
+		if err != nil {
+			logger.FromContext(ctx).Sugar().Warnf("deployments: row closing error: %v", err)
+		}
+	}(rows)
+	return processDeployment(rows)
+}
+
+func (h *DBHandler) DBSelectAllLatestDeploymentsForApplication(ctx context.Context, tx *sql.Tx, appName string) (map[string]Deployment, error) {
+	span, ctx := tracer.StartSpanFromContext(ctx, "DBSelectAllLatestDeploymentsForApplication")
+	defer span.Finish()
+	selectQuery := h.AdaptQuery(`
+		SELECT created, appname, releaseVersion, envName, metadata
+		FROM deployments
+		WHERE deployments.appname = (?) AND deployments.releaseVersion IS NOT NULL;
+	`)
+	span.SetTag("query", selectQuery)
+	rows, err := tx.QueryContext(
+		ctx,
+		selectQuery,
+		appName,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("could not select deployment from DB. Error: %w\n", err)
+	}
+	defer func(rows *sql.Rows) {
+		err := rows.Close()
+		if err != nil {
+			logger.FromContext(ctx).Sugar().Warnf("deployments: row closing error: %v", err)
+		}
+	}(rows)
+	return processAllLatestDeploymentsForApp(rows)
+}
+
+func (h *DBHandler) DBSelectAllLatestDeploymentsOnEnvironment(ctx context.Context, tx *sql.Tx, envName string) (map[string]*int64, error) {
+	span, ctx := tracer.StartSpanFromContext(ctx, "DBSelectAllLatestDeploymentsOnEnvironment")
+	defer span.Finish()
+	selectQuery := h.AdaptQuery(`
+		SELECT appname, releaseVersion
+		FROM deployments
+		WHERE deployments.envName= ?;
+	`)
+
+	span.SetTag("query", selectQuery)
+	rows, err := tx.QueryContext(
+		ctx,
+		selectQuery,
+		envName,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("could not select deployment from DB. Error: %w\n", err)
```

Comment:
please add a little more context here, e.g.
```suggestion
		return nil, fmt.Errorf("could not select deployment for env %s from DB. Error: %w\n", envName, err)
```

Reply:
Done